src/main/java/frc/robot/subsystems/Conveyor.java

```diff
@@ -57,94 +113,268 @@ public void setBackMotorPower(double power) {
+
+                } else if (beltState == ConveyorState.NONE
+                    && ((colorSensorMeasurement == "Red"
+                    && indexState == ConveyorState.RED
+                    && !isBallIndexed)
+                    || (colorSensorMeasurement == "Blue"
+                    && indexState == ConveyorState.BLUE
+                    && !isBallIndexed))) {
+                    // If that measurement matches indexState, the ball is not yet indexed, and
+                    // beltState is currently empty...
+
+                    indexState = ConveyorState.NONE;
+
+                }
+            }
+            
+            if (indexRPM > 0
+                && indexState != ConveyorState.NONE
+                && wasBallIndexed
+                && !isBallIndexed) {
+                // If we probably fired the indexed ball...
+
+                indexState = ConveyorState.NONE;
+
+            }
+
+        } else {
+
+            if (indexRPM > 0
+                && indexState != ConveyorState.NONE
+                && wasBallIndexed
+                && !isBallIndexed) {
+                // If the belt is stationary, but we probably still fired the indexed ball...
+
+                indexState = ConveyorState.NONE;
+
+            }
+        }
+    }
+
+    /**
+     * Returns the type of ball present in the position closer to the intake.
+     *
+     * @return the type of ball present in the position closer to the intake
+     */
+    public ConveyorState getBeltState() {
+        return beltState;
+    }
+
+    /**
+     * Returns the type of ball present in the position further away from the intake.
+     * Note: To keep the logic simple, the conveyor code automatically assigns the
+     * furthest-along ball to the indexed state, even if it has not gotten off the conveyor yet.
+     * Check isBallIndexed to see if the ball is actually ready.
+     *
+     * @return the type of ball present in the position further away from the intake
+     */
+    public ConveyorState getWillBeIndexedState() {
+        return indexState;
+    }
+
+    /**
+     * Returns whether or not there is a ball ready to be fired.
+     * This should be checked by the shooter.
+     *
+     * @return whether or not there is a ball ready to be fired
+     */
+    public boolean getIsBallIndexed() {
+        if (MIN_INDEXED_LIDAR_DISTANCE <= lidarDistance
+            && lidarDistance <= MAX_INDEXED_LIDAR_DISTANCE) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns whether or not there is a single ball in the conveyor.
+     * This should be checked by the shooter.
+     *
+     * @return whether or not there is a single ball in the conveyor
+     */
+    public boolean getDoesBallExist() {
+        if (MIN_EXISTS_LIDAR_DISTANCE <= lidarDistance
+            && lidarDistance <= MAX_EXISTS_LIDAR_DISTANCE) {
+            return true;
+        } else {
+            return false;
+        }
     }
 
```

Comment:
add a function to get lidar distance (might use it for calibrating robot pos in auto)